v0.8.2 release notes (#1222)
OSOceanAcoustics/echopype: docs/source/whats-new.md

See [GitHub releases page](https://github.com/OSOceanAcoustics/echopype/releases) for the complete history.

# v0.8.2 (2023 November 20)

## Overview

This release includes a number of important performance enhancements, bug fixes, and under-the-hood refactoring to improve code readability.

## Enhancements
- Full refactor of the functionality to directly storing parsed data to zarr before set_groups_* (#1185, #1218 [NOT MERGED], #1070)
  - This avoids memory expansion during NaN padding across channel
- Improve `compute_MVBS` using flox (#1124)
- Revive and improve `compute_NASC` using flox (#1167)

> - Refactor AZFP XML parser

Reviewer: ```suggestion
- Refactor AZFP XML parser
  - Improved XML parsing by using `xml.etree.ElementTree`, as used elsewhere in echopype (#1135)
```

Author: ```suggestion
- Refactor AZFP XML parser
  - Improved XML parsing by using `xml.etree.ElementTree` to be consistent with other XML parsers in echopype (#1135)
```

  - Parse more parameters and store in the resulting `EchoData` objects (#1135)

>   - Add support for AZFP multiple phase settings (#1182)

Reviewer: Based on your email exchange with Steve, it sounded like the support that was added would need additional work, right? If so, maybe add some qualifiers to this statement ("initial support", "partial support", ...)

>   - Parse more parameters and store in the resulting `EchoData` objects (#1135)
>   - Add support for AZFP multiple phase settings (#1182)

Author: Good call!
```suggestion
  - Parse more parameters and store in the resulting `EchoData` objects (#1135)
  - Add partial support for AZFP multiple phase settings (#1182)
```

> - AZFP pressure data handling
>   - Parse and store AZFP pressure data (#1189)
>   - Add `depth_from_pressure` method and store calculated depth data in Platform group (#1207, #1202 [NOT MERGED])

Reviewer: The `depth_from_pressure` function is a generic utility function and is not tied to AZFP. So, I think it should be its own item:

```suggestion
  - Add `depth_from_pressure` utility function to calculate depth from pressure and latitude (#1207)
- AZFP pressure data handling
  - Parse and store AZFP pressure data in the Environment group (#1189)
  - Use pressure to calculate depth, stored in `vertical_offset` in the Platform group (#1202) [NOT MERGED]
```

Author: I am a bit hesitate regarding converting pressure to depth directly, because of the uncertainty in latitude. I'll comment on it in #1202.

> - Remove unused mechanism to selectively parse some EK datagrams (#1214)

Reviewer: If we merge my PR #1226 (look it over when you have a chance), let's add a  new item above this one, with the scope of that PR: In AZFP Environment group, adding two missing variables (though likely filled with nan) and making the addition of temperature and pressure variables contingent on the presence of actual, valid data.

Author: I am not able to review those before Friday, so let's defer those to the next release

- Enhancements of `EchoData.update_platform` method
  - Factor out uility functions to a submodule (#1209)
  - Auto-assign timestamp for fixed-location external data using first `ping_time`(#1196)
- Add support for `consolidate` subpackage functions to accept both in-memory or stored datasets (#1216) [NOT MERGED]

## Bug fixes
- Fix `EchoData.to_zarr` encoding (#1128)
  - Eliminate preferred chunks
  - Ensuring chunk alignment and encoding only when handling dask arrays
- Fix scaling bugs in `compute_NASC` (#1167)
- Fix `UnicodeDecodeError` for ES60 files (#1215) [NOT MERGED]
- Handle missing `receiver_sampling_frequency` for EK80 data (#1219) [NOT MERGED]

## Infrastructure and refactoring
- Add support for running individual test files (#1166)
- Add module based testing (#1180)

> - Update CI to barebone python (#1192)

Reviewer: ```suggestion
- Update CI to barebone python, without conda (#1192)
```

See [GitHub releases page](https://github.com/OSOceanAcoustics/echopype/releases) for the complete history.

# v0.8.2 (2023 November 20)

## Overview

This release includes a number of important performance enhancements, bug fixes, and under-the-hood refactoring to improve code readability.

## Enhancements
- Full refactor of the functionality to directly storing parsed data to zarr before set_groups_* (#1185, #1218, #1070)
  - This avoids memory expansion during NaN padding across channel
- Improve `compute_MVBS` using flox (#1124)
- Revive and improve `compute_NASC` using flox (#1167)
- Refactor AZFP XML parser
  - Improved XML parsing by using `xml.etree.ElementTree` to be consistent with other XML parsers in echopype (#1135)
  - Parse more parameters and store in the resulting `EchoData` objects (#1135)
  - Add partial support for AZFP multiple phase settings (#1182)
- Overhaul AZFP Environment group
  - Set mandatory variables not in data to NaN (`sounds_speed_indicative`, `absorption_indicative`) (#1226)

>   - Parse and store AZFP pressure data when exist (#1189)

Author: ```suggestion
  - Parse and store AZFP pressure data when exist (#1189, #1226)
```

>   - Add `depth_from_pressure` method and store calculated depth data in Platform group (#1207, #1202 [NOT MERGED])

Author: ```suggestion
- Add `utils.misc.depth_from_pressure` function for converting pressure to depth (#1207)
```

- Remove unused mechanism to selectively parse some EK datagrams (#1214)
- Enhancements of `EchoData.update_platform` method

>   - Factor out uility functions to a submodule (#1209)

Reviewer: ```suggestion
  - Factor out utility functions to a submodule (#1209)
```
